src/SprykerEco/Zed/AvalaraTax/Business/Mapper/AvalaraTransactionResponseMapper.php

```diff
@@ -0,0 +1,123 @@
+    ): AvalaraCreateTransactionResponseTransfer {
+        $avalaraTransactionTransfer = (new AvalaraTransactionTransfer())->fromArray($this->convertStdClassToArray($transactionModel), true);
+        $avalaraTransactionTransfer
+            ->setAddresses($this->encodeStdClassPropertyToJson($transactionModel, static::KEY_ADDRESSES))
+            ->setLocationTypes($this->encodeStdClassPropertyToJson($transactionModel, static::KEY_LOCATION_TYPES))
+            ->setSummary($this->encodeStdClassPropertyToJson($transactionModel, static::KEY_SUMMARY))
+            ->setMessages($this->encodeStdClassPropertyToJson($transactionModel, static::KEY_MESSAGE))
+            ->setInvoiceMessages($this->encodeStdClassPropertyToJson($transactionModel, static::KEY_INVOICE_MESSAGE));
+
+        $avalaraTransactionLineTransfers = [];
+        foreach ($transactionModel->lines as $transactionLineModel) {
+            $avalaraTransactionLineTransfers[] = $this->mapTransactionLineModelToAvalaraTransactionLineTransfer(
+                $transactionLineModel,
+                new AvalaraTransactionLineTransfer()
+            );
+        }
+        $avalaraTransactionTransfer->setLines(new ArrayObject($avalaraTransactionLineTransfers));
+
+        return $avalaraCreateTransactionResponseTransfer->setTransaction($avalaraTransactionTransfer);
+    }
+
+    /**
+     * @param \stdClass|\Avalara\TransactionLineModel $transactionLineModel
+     * @param \Generated\Shared\Transfer\AvalaraTransactionLineTransfer $avalaraTransactionLineTransfer
+     *
+     * @return \Generated\Shared\Transfer\AvalaraTransactionLineTransfer
+     */
+    protected function mapTransactionLineModelToAvalaraTransactionLineTransfer(
+        stdClass $transactionLineModel,
+        AvalaraTransactionLineTransfer $avalaraTransactionLineTransfer
+    ): AvalaraTransactionLineTransfer {
+        $transactionLineModelData = $this->convertStdClassToArray($transactionLineModel);
+
+        $avalaraTransactionLineTransfer = $avalaraTransactionLineTransfer->fromArray($transactionLineModelData, true);
+        $avalaraTransactionLineTransfer
+            ->setDetails($this->encodeStdClassPropertyToJson($transactionLineModel, static::KEY_DETAILS))
+            ->setNonPassthroughDetails($this->encodeStdClassPropertyToJson($transactionLineModel, static::KEY_NON_PASSTHROUGH_DETAILS))
+            ->setLineLocationTypes($this->encodeStdClassPropertyToJson($transactionLineModel, static::KEY_LINE_LOCATION_TYPES))
+            ->setParameters($this->encodeStdClassPropertyToJson($transactionLineModel, static::KEY_PARAMETERS))
+            ->setTaxAmountByTaxTypes($this->encodeStdClassPropertyToJson($transactionLineModel, static::KEY_TAX_AMOUNT_BY_TAX_TYPES));
+
+        return $avalaraTransactionLineTransfer;
+    }
+
+    /**
+     * @param \stdClass $stdClass
+     *
+     * @return array
+     */
+    protected function convertStdClassToArray(stdClass $stdClass): array
```

Comment:
* Please check if you really need JSON manipulation.
* It looks like `(array)$stdClass` will do what you need.

Reply:
Resolved.